bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -164,51 +193,68 @@ pub mod pallet {
 			let asset_location: Location =
 				(*asset_id).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
 
-			let mut checked = false;
-			if asset_location.eq(&origin_location) || asset_location.starts_with(&origin_location) {
-				checked = true
-			}
-			ensure!(checked, <Error<T>>::OwnerCheck);
+			// Burn Ether Fee for the cost on ethereum
+			Self::burn_for_teleport(&origin_location, &(T::FeeAsset::get(), fee).into())?;
+
+			// Burn RemoteExecutionFee for the cost on bridge hub
+			Self::burn_for_teleport(&origin_location, &T::RemoteExecutionFee::get())?;
 
-			Self::burn_fees(origin_location.clone(), fee)?;
+			let reanchored_asset_location = asset_location
+				.clone()
+				.reanchored(&T::BridgeHubLocation::get(), &T::UniversalLocation::get())
+				.map_err(|_| Error::<T>::LocationConversionFailed)?;
 
-			let call = SnowbridgeControl::Control(ControlCall::RegisterToken {
-				asset_id: Box::new(VersionedLocation::from(asset_location.clone())),
+			let call = BridgeHubRuntime::EthereumSystem(EthereumSystemCall::RegisterToken {
+				asset_id: Box::new(VersionedLocation::from(reanchored_asset_location.clone())),
 				metadata,
 				fee,
 			});
 
+			let topic = blake2_256(&("SnowbridgeFrontend", call.clone()).encode());
+
 			let xcm: Xcm<()> = vec![
-				ReceiveTeleportedAsset(T::DeliveryFee::get().into()),
-				PayFees { asset: T::DeliveryFee::get() },
+				ReceiveTeleportedAsset(T::RemoteExecutionFee::get().into()),
+				PayFees { asset: T::RemoteExecutionFee::get() },
 				Transact {
 					origin_kind: OriginKind::Xcm,
 					call: call.encode().into(),
 					fallback_max_weight: None,
 				},
+				ExpectTransactStatus(MaybeErrorCode::Success),
+				SetTopic(topic),
 			]
 			.into();
 
-			Self::send(xcm)?;
+			let message_id = Self::send(origin_location.clone(), xcm)?;
 
-			Self::deposit_event(Event::<T>::RegisterToken { location: asset_location });
+			Self::deposit_event(Event::<T>::RegisterToken { location: asset_location, message_id });
 
 			Ok(())
 		}
 	}
 
 	impl<T: Config> Pallet<T> {
-		pub fn send(xcm: Xcm<()>) -> DispatchResult {
-			let bridgehub = Location::new(1, [Parachain(1002)]);
-			send_xcm::<T::XcmSender>(bridgehub, xcm).map_err(|_| Error::<T>::Send)?;
-			Ok(())
+		pub fn send(origin: Location, xcm: Xcm<()>) -> Result<H256, Error<T>> {
+			let bridgehub = T::BridgeHubLocation::get();
+			let (hash, price) =
```

Comment:
It doesn't have to be a hash. That's just an optional internal implementation detail.

```suggestion
			let (message_id, price) =
```